core/src/test/java/com/linecorp/armeria/client/endpoint/RestartableThreadTest.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client.endpoint;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestName;
+
+public class RestartableThreadTest {
+
+    @Rule
+    public final TestName testName = new TestName();
```

Comment:
nit: we can use `TestInfo` and convert this test using JUnit 5 if you want.

Reply:
👍 although I would have to rebase and I'm a little concerned bc the comments in this PR won't be trackable anymore + it's difficult for reviewers to see change.   If you think rebasing is fine, I can implement in this PR. let me know what you think:)

Reply:
Yeah, it's fine as it is. :-) We were planning to convert all the tests using JUnit 5 eventually, but we can change it later.